File: oo-workshop/single-responsibillity-principle/src/test/java/com/example/openclosed/mobprogramming/problem/OrderReportTest.java
package com.example.openclosed.mobprogramming.problem;

import org.junit.Test;;

import static org.junit.Assert.assertTrue;

class OrderReportTest {

    @Test
public void invoice() {
        final OrderReport given = new OrderReport("Adam Schnaare",1000);
        final String when = given.invoice();

        // then
        assertTrue(when.contains("Invoice"));
        assertTrue(when.contains("Adam Schnaare"));
        assertTrue(when.contains("1000"));
    }

    @Test
public void shippingLabel() {
        final OrderReport given = new OrderReport("Adam Schnaare",1000);
        final String when = given.shippingLabel();

        // then
        assertTrue(when.contains("Shipping Label"));
        assertTrue(when.contains("Adam Schnaare"));
    }
}

File: oo-workshop/tsexercises/tests/unit/openclosed/orderReport.test.ts
import {OrderReport} from '../../../src/openclosed/orderreport'

describe('order report', () => {
    it('should create an invoice', () => {
        const given: OrderReport = new OrderReport("Adam Schnaare", 1000)
        const when: string = given.invoice()
  
        // then
        expect(when).toContain("Invoice")
        expect(when).toContain("Adam Schnaare")
        expect(when).toContain("1000")
    })

    it('should create a shipping label', () => {
        const given: OrderReport = new OrderReport("Adam Schnaare", 1000)
        const when: string = given.invoice()

        // then
        expect(when).toContain("Shipping Label")
        expect(when).toContain("Adam Schnaare")
  })
})